tests/unit_tests/lapack/source/gebrd.cpp

```diff
@@ -76,6 +76,7 @@ bool accuracy(const sycl::device& dev, int64_t m, int64_t n, int64_t lda, uint64
         TEST_RUN_LAPACK_CT_SELECT(
             queue, scratchpad_size = oneapi::mkl::lapack::gebrd_scratchpad_size<fp>, m, n, lda);
 #endif
+        queue.wait_and_throw();
```

Comment:
Since the scratchpad_size functions return the scratchpad_size value (rather than an event), that seems to make them synchronous, and so is it required to wait on the queue?

Reply:
Yeah these queue.wait_and_throw() should be unnecessary, thanks for pointing it out. This made me realize that actually I can still improve the implementation for scratchpad functions, since in the cuda backend they are asynchronous. I will update them soon.

Reply:
I've now reverted all changes to these tests. I've also improved the synchronous implementation of scratchpad_size functions in this commit: https://github.com/oneapi-src/oneMKL/pull/572/commits/c082d4a0a90bfa3ecd4a144bd9cb4b0f5adb54cc

I swapped the queue.wait() with a cuStreamSync call that waits only on the stream that is used, in addition to waiting on the event returned from the host_task (in the case that it returns before cuStreamSync has been returned). This means that we don't unnecessarily sync all streams in the queue anymore.